packages/api/tests/models/granules/test-granule-reporting.js

```diff
@@ -23,214 +23,303 @@ test.before(async (t) => {
   const { granuleModel } = t.context;
 
   const granule = fakeGranuleFactoryV2({ status: 'failed' });
 
-  await granuleModel._validateAndStoreGranuleRecord(granule);
+  await granuleModel._storeGranuleRecord(granule);
 
   const updatedGranule = {
     ...granule,
     status: 'running',
   };
 
-  await granuleModel._validateAndStoreGranuleRecord(updatedGranule);
+  await t.throwsAsync(granuleModel._storeGranuleRecord(updatedGranule));
 
   const fetchedItem = await granuleModel.get({ granuleId: granule.granuleId });
 
   t.is(fetchedItem.status, 'failed');
 });
 
-test('_validateAndStoreGranuleRecord() will allow a running status to replace a completed status for a new execution', async (t) => {
+test('_storeGranuleRecord() will allow a running status to replace a completed status for a new execution', async (t) => {
   const { granuleModel } = t.context;
 
   const granule = fakeGranuleFactoryV2();
 
-  await granuleModel._validateAndStoreGranuleRecord(granule);
+  await granuleModel._storeGranuleRecord(granule);
 
   const updatedGranule = {
     ...granule,
     execution: 'new-execution-url',
     status: 'running',
   };
 
-  await granuleModel._validateAndStoreGranuleRecord(updatedGranule);
+  await granuleModel._storeGranuleRecord(updatedGranule);
 
   const fetchedItem = await granuleModel.get({ granuleId: granule.granuleId });
 
   t.is(fetchedItem.status, 'running');
 });
 
-test('_validateAndStoreGranuleRecord() will allow a running status to replace a failed status for a new execution', async (t) => {
+test('_storeGranuleRecord() will allow a running status to replace a failed status for a new execution', async (t) => {
   const { granuleModel } = t.context;
 
   const granule = fakeGranuleFactoryV2({ status: 'failed' });
 
-  await granuleModel._validateAndStoreGranuleRecord(granule);
+  await granuleModel._storeGranuleRecord(granule);
 
   const updatedGranule = {
     ...granule,
     execution: 'new-execution-url',
     status: 'running',
   };
 
-  await granuleModel._validateAndStoreGranuleRecord(updatedGranule);
+  await granuleModel._storeGranuleRecord(updatedGranule);
 
   const fetchedItem = await granuleModel.get({ granuleId: granule.granuleId });
 
   t.is(fetchedItem.status, 'running');
 });
 
-test('_validateAndStoreGranuleRecord() does not throw an error for a failing record', async (t) => {
+test('_validateAndStoreGranuleRecord() does throw validation error', async (t) => {
   const { granuleModel } = t.context;
 
   const granule = fakeGranuleFactoryV2();
   // granule without granuleId should fail validation
   delete granule.granuleId;
 
-  try {
-    await granuleModel._validateAndStoreGranuleRecord(granule);
-    t.pass();
-  } catch (error) {
-    t.fail(`Expected error not to be thrown, caught: ${error}`);
-  }
+  await t.throwsAsync(granuleModel._validateAndStoreGranuleRecord(granule));
+});
+
+test('_validateAndStoreGranuleRecord() throws error for a conditional check exception', async (t) => {
```

Comment:
It's not clear from a casual read what this is actually doing - at first glance it looks like the test is checking that you can't update a failed record to running.   ALso if we want to be extra certain, we could check the error type. 

Something more like: 

```suggestion
test('_validateAndStoreGranuleRecord() throws an error if replacing granule without updated execution', async (t) => {
```

Reply:
8fa4866781a2269eaf70379c0d59cd30822e196d